src/build_system.ml

```diff
@@ -1553,7 +1568,7 @@ end = struct
             Fiber.return ()
         end
       and proc_rule dep =
-        get_rule_other t dep >>= function
+        Fiber.return (get_rule_other t dep) >>= function
```

Comment:
Rather than add the `Fiber.return`, you can just pattern match on `get_rule_other` directly`.

Reply:
done